### PR TITLE
add AKS example

### PR DESCRIPTION
**Pull request by seanknox**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use resourceGroup().location for resource locations
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/bp-checklist.md

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

* Add AKS (Managed Kubernetes) example

### CODE REVIEW

File: 101-aks/README.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Managed Kubernetes (AKS)
```
Comment (bmoore-msft) [inline 101-aks/README.md]: add a proper readme - check the sample or another in the repo

File: 101-aks/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+			"metadata": {
+				"description": "The number of nodes for the cluster."
+			},
+			"minValue": 1,
+			"maxValue": 50
+		},
+		"agentVMSize": {
+			"type": "string",
+			"defaultValue": "Standard_D2_v2",
+			"allowedValues": [
```
Comment (bmoore-msft) [inline 101-aks/azuredeploy.json]: remove allowed values

File: 101-aks/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+				"Standard_G1",
+				"Standard_G2"
+			],
+			"metadata": {
+				"description": "The size of the Virtual Machine."
+			}
+		},
+		"linuxAdminUsername": {
+			"type": "string",
+			"defaultValue": "azureuser",
```
Comment (bmoore-msft) [inline 101-aks/azuredeploy.json]: remove defaultValue - increases surface area

File: 101-aks/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+			"type": "string",
+			"metadata": {
+				"description": "Configure all linux machines with the SSH RSA public key string. Your key should include three parts, for example 'ssh-rsa AAAAB...snip...UcyupgH azureuser@linuxvm'"
+			}
+		},
+		"servicePrincipalClientId": {
+			"metadata": {
+				"description": "Client ID (used by cloudprovider)"
+			},
+			"type": "securestring",
```
Comment (bmoore-msft) [inline 101-aks/azuredeploy.json]: this doesn't need to be secure

File: 101-aks/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+			"defaultValue": "n/a"
```
Comment (bmoore-msft) [inline 101-aks/azuredeploy.json]: remove defaultValue if it's not valid

File: 101-aks/azuredeploy.json [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+		},
+		"servicePrincipalClientSecret": {
+			"metadata": {
+				"description": "The Service Principal Client Secret."
+			},
+			"type": "securestring",
+			"defaultValue": "n/a"
```
Comment (bmoore-msft) [inline 101-aks/azuredeploy.json]: remove defaultValue

File: 101-aks/azuredeploy.json [hunk 7]
```diff
@@ -0,0 +1,206 @@
...
+		},
+		"osType": {
```
Comment (bmoore-msft) [inline 101-aks/azuredeploy.json]: if there's only one allowed value then we can remove the param and use a literal or var

File: 101-aks/azuredeploy.json [hunk 8]
```diff
@@ -0,0 +1,206 @@
...
+			"allowedValues": [
+				"1.7.7",
+				"1.8.1"
+			],
+			"metadata": {
+				"description": "The version of Kubernetes."
+			}
+		}
+	},
+	"variables": {
```
Comment (bmoore-msft) [inline 101-aks/azuredeploy.json]: remove the variables and just use the parameters

File: 101-aks/azuredeploy.json [hunk 9]
```diff
@@ -0,0 +1,206 @@
...
+					"ClientId": "[parameters('servicePrincipalClientId')]",
+					"Secret": "[parameters('servicePrincipalClientSecret')]"
+				}
+			}
+		}
+	],
+	"outputs": {
+		"controlPlaneFQDN": {
+			"type": "string",
+			"value": "[reference(concat('Microsoft.ContainerService/managedClusters/', variables('resourceName'))).fqdn]"
```
Comment (bmoore-msft) [inline 101-aks/azuredeploy.json]: simplify - just use:

reference(variables('resourceName')).fqdn

File: 101-aks/azuredeploy.parameters.json [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+			"value": "GEN-UNIQUE"
+		},
+		"agentCount": {
+			"value": 1
+		},
+		"agentVMSize": {
+			"value": "Standard_D2_v2"
+		},
+		"linuxAdminUsername": {
+			"value": "azureuser"
```
Comment (bmoore-msft) [inline 101-aks/azuredeploy.parameters.json]: GEN-UNIQUE

File: 101-aks/azuredeploy.parameters.json [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+		},
+		"sshRSAPublicKey": {
+			"value": "GEN-UNIQUE"
```
Comment (bmoore-msft) [inline 101-aks/azuredeploy.parameters.json]: GEN-SSH-PUB-KEY

File: 101-aks/azuredeploy.parameters.json [hunk 12]
```diff
@@ -0,0 +1,27 @@
+{
+	"$schema": "http://schema.management.azure.com/schemas/2015-01-01/deploymentParameters.json#",
+	"contentVersion": "1.0.0.0",
+	"parameters": {
+		"dnsPrefix": {
+			"value": "GEN-UNIQUE"
+		},
+		"agentCount": {
```
Comment (bmoore-msft) [inline 101-aks/azuredeploy.parameters.json]: remove the params that have defaultValues

File: 101-aks/azuredeploy.parameters.json [hunk 13]
```diff
@@ -0,0 +1,27 @@
...
+			"value": "Standard_D2_v2"
+		},
+		"linuxAdminUsername": {
+			"value": "azureuser"
+		},
+		"sshRSAPublicKey": {
+			"value": "GEN-UNIQUE"
+		},
+		"servicePrincipalClientId": {
+			"value": "GEN-UNIQUE"
```
Comment (bmoore-msft) [inline 101-aks/azuredeploy.parameters.json]: GEN-GUID

File: 101-aks/metadata.json [hunk 14]
```diff
@@ -0,0 +1,7 @@
+{
+  "itemDisplayName": "AKS",
+  "description": "Azure Container Service preview (AKS) is the quickest path from zero to Kubernetes on Azure. This new service features an Azure-hosted control plane, automated upgrades, self-healing, easy scaling, and a simple user experience for both developers and cluster operators. With AKS, customers get the benefits of open source Kubernetes without complexity and operational overhead.\nTo help you get started with AKS, the managed Kubernetes service is free. With AKS you only pay for your Kubernetes nodes. This means on Azure, you will pay nothing for the management of your Kubernetes cluster, ever.\nAKS is currently available for Standard A, D, DS, G, GS, F, and FS series Linux virtual machines. You are only charged for the compute instances you choose, as well as the other underlying infrastructure resources consumed such as storage and networking. There are no incremental charges for AKS itself.\nDeployments typically takes a few minutes. Once complete, you can access and manage your cluster through \"az aks\" and \"kubectl\".",
```
Comment (bmoore-msft) [inline 101-aks/metadata.json]: check the CI failure, this is too long

File: 101-aks/metadata.json [hunk 15]
```diff
@@ -0,0 +1,7 @@
+{
+  "itemDisplayName": "AKS",
```
Comment (bmoore-msft) [inline 101-aks/metadata.json]: need more detail here

File: 101-aks/README.md (hunk 1 above)
Comment (seanknox) [reply 101-aks/README.md]: I followed the existing example in 101-acs.

File: 101-aks/azuredeploy.json (hunk 7 above)
Comment (seanknox) [reply 101-aks/azuredeploy.json]: Multiple OS types will be arriving soon.

**Comment by seanknox** [comment]
Thanks for the feedback. Think I addressed them.

File: 101-aks/README.md (hunk 1 above)
Comment (bmoore-msft) [reply 101-aks/README.md]: Use this one: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/sample-README.md

File: 101-aks/azuredeploy.json [hunk 16]
```diff
@@ -0,0 +1,130 @@
...
+				"description": "The version of Kubernetes."
+			}
+		}
+	},
+	"resources": [
+		{
+			"apiVersion": "2017-08-31",
+			"type": "Microsoft.ContainerService/managedClusters",
+			"location": "[resourceGroup().location]",
+			"name": "[variables('resourceName')]",
```
Comment (bmoore-msft) [inline 101-aks/azuredeploy.json]: Need to replace vars w/ param here - check the CI failure

**Comment by bmoore-msft** [comment]
ok - thx... just need to address the readme and the CI failure (param/var use)

**Comment by bmoore-msft** [comment]
Closing abandoned request - please reopen if you can address the feedback.

**Comment by seanknox** [comment]
@squillace Do you know of someone who could pick these quickstart examples for AKS? It's close but needs to address the issues Brian pointed out.

cc @slack

**Comment by squillace** [comment]
@seanknox sure. Was this a request for a new template? catching up here.

**Comment by seanknox** [comment]
@squillace yep, new AKS example is in this branch.

**Comment by squillace** [comment]
I pinged Saurya about it, let's see who can pick it up. I'm not just underwater this week, I'm under the outer layers of the earth's crust.

**Pull request closed without merging**